packages/@headlessui-react/src/components/combobox/combobox.tsx

```diff
@@ -537,9 +539,13 @@ function VirtualProvider(props: {
           }
 
           // Scroll to the active index
-          if (data.activeOptionIndex !== null && options.length > data.activeOptionIndex) {
-            virtualizer.scrollToIndex(data.activeOptionIndex)
-          }
+          //
+          // Workaround for: https://github.com/TanStack/virtual/issues/879
+          d.nextFrame(() => {
+            if (data.activeOptionIndex !== null && options.length > data.activeOptionIndex) {
+              virtualizer.scrollToIndex(data.activeOptionIndex)
+            }
+          })
```

Comment:
Not gonna hold the PR up on this question but is it _valid_ to hoist the if outside the next frame call or nah?

Reply:
Yes if we track the variables because `data` could change in the meantime. I first did that and TypeScript gave me:
```
Argument of type 'number | null' is not assignable to parameter of type 'number'.   Type 'null' is not assignable to type 'nu
mber'. (tsserver 2345)
```